interactions/utils/abc/base_iterators.py

```diff
@@ -0,0 +1,97 @@
+from abc import ABCMeta, abstractmethod
+from math import inf
+from typing import TYPE_CHECKING, Callable, List, Optional, TypeVar, Union
+
+from ..missing import MISSING
+
+_T = TypeVar("_T")
+_O = TypeVar("_O")
+
+__all__ = ("BaseAsyncIterator", "BaseIterator")
+
+if TYPE_CHECKING:
+    from ...api.http.client import HTTPClient
+    from ...api.models.misc import Snowflake
+
+
+class BaseAsyncIterator(metaclass=ABCMeta):
```

Comment:
Can you add a flatten method to completely consume the generator and return the list of results?